best_practices/production_deployments/plugins/WITHOUT_COMPILERS.md

```diff
@@ -0,0 +1,138 @@
+# Pre-compiling native extensions for Sensu Plugins gem dependencies
+
```

Comment:
Can you add a short paragraph on the topic? I don't think this is terribly approachable to people. Something simple like: "When trying to reduce the number of moving parts in a production deployment of anything, compiled code is preferred over interpreted code. There are some good references at the bottom (link to bottom) Here's how the Sensu community recommends getting to that state." 

Then at the bottom there could be 

> ## Follow up reading
> 
> Here are references that will elaborate on why this is a great idea: 
> * Foo on CI/CD philosophy

Etc

Reply:
Well this is a fairly complex issue and chances are if you are coming here you already know the benefits even if not the how of doing so. As I think about how this fits into the directory of best practices I do think that what you are saying makes a lot of sense as this is something that is great for new users of sensu to flip through before launching sensu in production. I need to spend some time to really do it justice. This is especially true as most of the time that people want to do this they usually have a bad reason (almost always security) and are familiar enough enough to know why we are jumping through these hoops. Let's get the rest of the PR ready and then we can talk about this as I am sure I will ruffle a few feathers and I'd like to keep git history as clean as possible.

Reply:
Sure, this is definitely not a prereq to merge. Having a lightweight intro is a good start.

Reply:
I have my "lightweight" explanation added...